test/Window/WindowBase.test.cpp

```diff
@@ -104,14 +104,29 @@ TEST_CASE("[Window] sf::WindowBase", runDisplayTests())
 
     SECTION("pollEvent()")
     {
-        sf::WindowBase windowBase;
-        CHECK(!windowBase.pollEvent());
+        SECTION("Uninitialized window")
+        {
+            sf::WindowBase windowBase;
+            CHECK(!windowBase.pollEvent());
+        }
     }
 
     SECTION("waitEvent()")
     {
-        sf::WindowBase windowBase;
-        CHECK(!windowBase.waitEvent());
+        SECTION("Uninitialized window")
+        {
+            sf::WindowBase windowBase;
+            CHECK(!windowBase.waitEvent());
+        }
+
+        SECTION("Initialized window")
+        {
+            sf::WindowBase windowBase(sf::VideoMode({360, 240}), "WindowBase Tests");
+
+            // Wait for an event with a timeout, doesn't matter if it is received or not,
+            // the purpose of this code is to test the timeout itself
+            [[maybe_unused]] const sf::Event event = windowBase.waitEvent(sf::milliseconds(1));
```

Comment:
I think we can take these tests further. How about we measure how long the function takes to return and ensure it takes less than the given timeout value with some margin of error. The margin of error will take some consideration. Here's what I came up with. Unfortunately I still get occasional failures so these tests are too flaky in their current state. Hopefully we can find a more robust way of doing this.

```cpp
        SECTION("Initialized window")
        {
            sf::WindowBase windowBase(sf::VideoMode({360, 240}), "WindowBase Tests");

            constexpr auto timeout = sf::milliseconds(200);
            const auto     now     = std::chrono::steady_clock::now();
            const auto     event   = windowBase.waitEvent(timeout);
            const auto     elapsed = std::chrono::steady_clock::now() - now;
            REQUIRE(elapsed < (timeout + sf::milliseconds(10)).toDuration());
            if (elapsed <= timeout.toDuration())
                CHECK(event);
            else
                CHECK(!event);
        }
```

Reply:
I'm going to try this out, I like the idea.